 Originally OpenWRT SVN changeset #1571 - bluetooth fixes from Tomas Vanek
 
  After this patch mtu can be greater than max packet size!
  Fixed by Leonid Lisovskiy <lly.dev@gmail.com>
 
--- linux-2.4.37/drivers/bluetooth/hci_usb.c	2008-12-02 11:01:34.000000000 +0300
+++ linux/drivers/bluetooth/hci_usb.c	2009-01-25 15:45:52.000000000 +0300
@@ -256,9 +256,12 @@ static int hci_usb_isoc_rx_submit(struct
 	struct _urb *_urb;
 	struct urb *urb;
 	int err, mtu, size;
-	void *buf;
+	void __attribute__((aligned(2))) *buf;
 
 	mtu  = husb->isoc_in_ep->wMaxPacketSize;
+#ifdef CONFIG_BCM4710
+	mtu &= ~1;		/* brcm: isoc buffers must be aligned on word boundary */
+#endif
         size = mtu * HCI_MAX_ISOC_FRAMES;
 
 	buf = kmalloc(size, GFP_ATOMIC);
@@ -279,6 +282,7 @@ static int hci_usb_isoc_rx_submit(struct
 	urb->dev      = husb->udev;
 	urb->pipe     = usb_rcvisocpipe(husb->udev, husb->isoc_in_ep->bEndpointAddress);
 	urb->complete = hci_usb_rx_complete;
+	urb->interval = 1;
 
 	urb->transfer_buffer_length = size;
 	urb->transfer_buffer = buf;
